src/main/scala/cognite/spark/v1/wdl/WdlModels.scala

```diff
@@ -0,0 +1,129 @@
+      WdlRetrieveDefinition("WellSource", "wellsources/list")
+    ),
+    WdlModel(
+      "wells",
+      None,
+      WdlRetrieveDefinition(
+        "Well",
+        "wells/list",
+        transformBody = jsonObject => jsonObject.add("includeWellbores", Json.fromBoolean(false))
+      )
+    ),
+    WdlModel(
+      "wellboresources",
+      Some(WdlIngestDefinition("WellboreSource", "wellbores")),
+      WdlRetrieveDefinition("WellboreSource", "wellboresources/list")
+    ),
+    WdlModel(
+      "wellbores",
+      None,
+      WdlRetrieveDefinition("Wellbore", "wellbores/list")
+    ),
+    WdlModel(
+      "depthmeasurements",
+      Some(WdlIngestDefinition("DepthMeasurementIngestion", "measurements/depth")),
+      WdlRetrieveDefinition("DepthMeasurement", "measurements/depth/list")
+    ),
+    WdlModel(
+      "timemeasurements",
+      Some(WdlIngestDefinition("TimeMeasurementIngestion", "measurements/time")),
+      WdlRetrieveDefinition("TimeMeasurement", "measurements/time/list")
+    ),
+    WdlModel(
+      "rigoperations",
+      Some(WdlIngestDefinition("RigOperationIngestion", "rigoperations")),
+      WdlRetrieveDefinition("RigOperation", "rigoperations/list")
+    ),
+    WdlModel(
+      "holesections",
+      Some(WdlIngestDefinition("HoleSectionGroupIngestion", "holesections")),
+      WdlRetrieveDefinition("HoleSectionGroup", "holesections/list")
+    ),
+    WdlModel(
+      "welltops",
+      Some(WdlIngestDefinition("WellTopGroupIngestion", "welltops")),
+      WdlRetrieveDefinition("WellTopGroup", "welltops/list")
+    ),
+    WdlModel(
+      "nptevents",
+      Some(WdlIngestDefinition("NptIngestion", "npt")),
+      WdlRetrieveDefinition("Npt", "npt/list")
+    ),
+    WdlModel(
+      "ndsevents",
+      Some(WdlIngestDefinition("NdsIngestion", "nds")),
+      WdlRetrieveDefinition("Nds", "nds/list")
+    ),
+    WdlModel(
+      "casings",
+      Some(WdlIngestDefinition("CasingSchematicIngestion", "casings")),
+      WdlRetrieveDefinition("CasingSchematic", "casings/list")
+    ),
+    WdlModel(
+      "trajectories",
+      Some(WdlIngestDefinition("TrajectoryIngestion", "trajectories")),
+      WdlRetrieveDefinition("Trajectory", "trajectories/list")
+    )
+  )
+
+  private val retrievalSchemaNames: Seq[String] = models.map(_.retrieve.schemaName)
+  private val schemaNamesForIngestion: Seq[String] = models.flatMap(_.ingest).map(_.schemaName)
+  private val shortNames: Seq[String] = models.map(_.shortName)
+
+  def fromRetrievalSchemaName(schemaName: String): WdlModel =
+    models
+      .find(p => p.retrieve.schemaName == schemaName)
+      .getOrElse(
+        throw new CdfSparkException(
+          s"Invalid schema name: `$schemaName`. The valid options are: [${retrievalSchemaNames.mkString(", ")}]")
+      )
+
+  def fromIngestionSchemaName(schemaName: String): WdlModel =
+    models
+      .find(p => p.ingest.map(_.schemaName).contains(schemaName))
```

Comment:
It is `Option.contains`. It will check if the _option_ contains the value, not if there is a sub string that matches.
```scala
Some("WellSource").contains("Well") // false
```